### PR TITLE
Command to listen for lease creation and send manifest automatically

### PR DESCRIPTION
**Pull request by jackzampolin**
I'm having an issue seeing the events. Nothing is printing after transaction gets sent. Repro for testing (from /_run/kube):

T1:  KIND_CONFIG=kind-config-80.yaml make kind-cluster-delete kind-cluster-create
T2: make clean init node-run
T3: make provider-create provider-run
T4: make bins && ../../akashctl --home ./cache/client/ tx deployment make-it-so deployment.yaml --from main

### CODE REVIEW

File: x/deployment/client/cli/tx.go [hunk 1]
```diff
@@ -32,10 +40,152 @@ func GetTxCmd(key string, cdc *codec.Codec) *cobra.Command {
...
+								&manifest.SubmitRequest{
+									Deployment: msg.ID.DeploymentID(),
+									Manifest:   mani,
+								},
+							); err != nil {
+								return err
+							}
+
+							if leases = leases - 1; leases == 0 {
+								fmt.Printf("Leases left %d...\n", leases)
```
Comment (jackzampolin) [inline x/deployment/client/cli/tx.go:148]: Planning on exiting the command here, but unable to make it exit cleanly with the current code.

Comment (jackzampolin) [reply x/deployment/client/cli/tx.go:149]: have this exiting cleanly!

File: x/deployment/client/cli/tx.go [hunk 2]
```diff
@@ -32,10 +40,154 @@ func GetTxCmd(key string, cdc *codec.Codec) *cobra.Command {
...
+			leases := len(groups)
+			group.Go(func() error {
+				for {
+					select {
+					case <-sub.Done():
+						return nil
+					case ev := <-sub.Events():
+						switch msg := ev.(type) {
+						case types.EventDeploymentCreated:
+							fmt.Printf("Deployment %d created...\n", msg.ID.DSeq)
```
Comment (Ropes) [inline x/deployment/client/cli/tx.go:121]: Could we use the ctx logger instead of fmt.Print?

File: x/deployment/client/cli/tx.go [hunk 3]
```diff
@@ -32,10 +40,154 @@ func GetTxCmd(key string, cdc *codec.Codec) *cobra.Command {
...
+					return err
+				}
+			}
+
+			// create a new pubsub bus to handle chain events
+			bus := pubsub.NewBus()
+			defer bus.Close()
+
+			// Create an error group to handle chain listener
+			group, cctx := errgroup.WithContext(ccontext.Background())
```
Comment (Ropes) [inline x/deployment/client/cli/tx.go:97]: :heart:

File: x/deployment/client/cli/tx.go [hunk 4]
```diff
@@ -32,10 +40,154 @@ func GetTxCmd(key string, cdc *codec.Codec) *cobra.Command {
...
+			if err != nil {
+				return err
+			}
+
+			// Default DSeq to the current block height
+			if id.DSeq == 0 {
+				if id.DSeq, err = currentBlockHeight(ctx); err != nil {
+					return err
+				}
+			}
```
Comment (Ropes) [inline x/deployment/client/cli/tx.go:90]: We should wrap all the Msg creation including the 
```
			msg := types.MsgCreateDeployment{
				ID: id,
				// TODO: Version:  []byte{0x1, 0x2},
				Groups: make([]types.GroupSpec, 0, len(groups)),
			}
```
 into a utility function. It's also called in normal Create, and it'd be good not to have drift.

File: x/deployment/client/cli/tx.go [hunk 5]
```diff
@@ -32,10 +40,154 @@ func GetTxCmd(key string, cdc *codec.Codec) *cobra.Command {
...
 		cmdGroupClose(key, cdc),
+		makeItSoCommand(key, cdc),
 	)...)
 	return cmd
 }
 
+func makeItSoCommand(key string, cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "make-it-so [sdl-file]",
+		Short: fmt.Sprintf("Create %s", key),
```
Comment (Ropes) [inline x/deployment/client/cli/tx.go:51]: Still `Create deployment`, might want to add a little explanation here for the command.

**Comment by jackzampolin** [comment]
Naming could def use some work 🤣 Also based on the conversation earlier today I think functionality similar to this could be blown out into a whole seperate `client-services` binary.

File: x/deployment/client/cli/tx.go (hunk 2 above)
Comment (jackzampolin) [reply x/deployment/client/cli/tx.go:121]: Thats a much better idea.

File: x/deployment/client/cli/tx.go (hunk 4 above)
Comment (jackzampolin) [reply x/deployment/client/cli/tx.go:90]: Agree, I'll add this.

File: x/deployment/client/cli/tx.go (hunk 2 above)
Comment (jackzampolin) [reply x/deployment/client/cli/tx.go:121]: So turns out you can't get a logger from the `ctx` 🤦

Comment (boz) [reply x/deployment/client/cli/tx.go:121]: the sdk client context - `ctx.PrintOutput()` maybe - it formats in json or yaml depending on flags or whatever.

Comment (jackzampolin) [reply x/deployment/client/cli/tx.go]: Yeah I want this to look more "log"-y. Think I'll emulate whats in the provider daemon.

File: deploy/deploy.go [hunk 6]
```diff
@@ -128,7 +128,7 @@ func CMD(cdc *codec.Codec) *cobra.Command {
 								return err
 							}
 
-							if leases = leases - 1; leases == 0 {
+							if leases--; leases == 0 {
```
Comment (troian) [inline deploy/deploy.go:131]: 😍

**Comment by jackzampolin** [comment]
Built this as a separate binary too if you want to give that a try: https://github.com/jackzampolin/deploy

**Pull request closed without merging**